src/test/kotlin/com/epirus/local/ledger/LocalLedgerTest.kt

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2020 Web3 Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.epirus.local.ledger
+
+import com.epirus.local.cli.Account
+import com.epirus.local.server.Request
+import com.epirus.local.server.RequestHandler
+import org.web3j.abi.FunctionEncoder
+import org.web3j.abi.datatypes.Function
+import org.web3j.abi.TypeReference
+import org.web3j.crypto.Credentials
+import org.web3j.crypto.RawTransaction
+import org.web3j.crypto.TransactionEncoder
+import org.web3j.utils.Numeric
+import java.io.File
+import java.math.BigInteger
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+
+class LocalLedgerTest {
+
+    private val accounts: List<Account> = generateAccounts()
+    private val genesis: String
+    private val localLedger: LocalLedger
+    private val requestHandler: RequestHandler
+    init {
+        genesis = createGenesis("src/test/resources/", accounts)
```

Comment:
this is a test artifact - better put it in "/build/test/tmp"
or use junit extension `@TempDir`

you can google about using temporary directories in junit 5 test

Reply:
maybe @cfelde can suggest something

Reply:
Worth having a look at how we did temp files in the web3j-cli, example: https://github.com/web3j/web3j-cli/blob/0e33382520e3d77227694e5ad870c80cc8a154b7/src/test/java/org/web3j/console/project/java/JavaProjectCreatorCLIRunnerTest.java#L27

Used to have @TempDir in there, but it was causing issues on Mac OS, likely related to the security model in Catalina.

Reply:
should i copy the Folders class or add a dependency to web3j-cli ?

Reply:
No dependency on the cli, that would not be appropriate. You can copy the class or make your own, up to you.

Reply:
Will do